search/api.py

```diff
@@ -32,6 +33,21 @@ def execute_search(search_obj):
     return search_obj.execute()
 
 
+def get_searchable_programs(user):
+    """
+    Determines the programs a user is eligible to search
+
+    Args:
+        user (django.contrib.auth.models.User): the user that is searching
+
+    Returns:
+        set(courses.models.Program): set of programs the user can search in
+    """
+    programs = list(get_advance_searchable_programs(user))
```

Comment:
Decided against inlining it because there's some very `Role`-specific logic in there, instead refactored `get_advance_searchable_programs` to return ids so we can do this method with 1 query.